### PR TITLE
Feature/bitmarket support

### PR DESCRIPTION
**Pull request by Marchuck**
I am not sure about bitMarket implemetation...

### CODE REVIEW

**Review by PawelDedio — APPROVED** [review]

File: src/main/kotlin/org/consoletrader/rx/CandleServiceObservable.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package org.consoletrader.rx
+
+import io.reactivex.Observable
+import org.consoletrader.candles.Candle
+import java.util.concurrent.atomic.AtomicReference
+
+object CandleServiceObservable {
```
Comment (tomaszbabiuk) [inline src/main/kotlin/org/consoletrader/rx/CandleServiceObservable.kt]: Nice work but we don't need that anymore. After ta4j is introduced that class is not required. We're putting all data to ta4j ticks collection and indicators logic is traversing the list of candles now.

File: src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class BitmarketCandleService : BaseApi<BitmarketPublicApi>(
+        anApi = BitmarketPublicApi::class.java,
+        endpoint = "https://www.bitmarket.pl/"),
+        CandlesService {
+
+    override fun getCandles(pair: CurrencyPair): Observable<Candle> {
+
+        //todo: pair is correct???
+        return getApi().queryCandles(pair = pair.toString())
+                .map {
```
Comment (tomaszbabiuk) [inline src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt]: CurrencyPair.toString() will return BTC/PLN but bit market is expecting BTCPLN (no slash). You probably need to glue this "${pair.base}${pair.counter}"

File: src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketPublicApi.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package org.consoletrader.candles.bitmarket
+
+import io.reactivex.Observable
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface BitmarketPublicApi {
+    @GET("/json/{pair}/ticker.json")
```
Comment (tomaszbabiuk) [inline src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketPublicApi.kt:8]: It's a wrong endpoint. This one is a ticker (no historic data) price with no time. We need to use this: https://www.bitmarket.pl/graphs/BTCPLN/1d.json
It's the closest we can get (returning candles every 16 minutes)
We also need to skip every second candle to be as close to 30 minutes as possible (other exchanges data).

File: src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                    val timestamp = Instant.EPOCH.epochSecond
```
Comment (tomaszbabiuk) [inline src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt]: You will have correct data for timeout when using correct endpoint.

File: src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                    //todo: review if this data is correct!!!
+                    Candle(timestamp = timestamp,
```
Comment (tomaszbabiuk) [inline src/main/kotlin/org/consoletrader/candles/bitmarket/BitmarketCandleService.kt]: Candle is removed. It's replaced by Tick from ta4j (latest commit on dev)

**Pull request merged**